File: src/network/packet/impl/PlayerTransferPacket.php
<?php

namespace pocketcloud\cloud\network\packet\impl;

use pocketcloud\cloud\network\client\ServerClient;
use pocketcloud\cloud\network\packet\ClientboundPacket;
use pocketcloud\cloud\network\packet\CloudboundPacket;
use pocketcloud\cloud\network\packet\CloudPacket;
use pocketcloud\cloud\network\packet\util\PacketData;
use pocketcloud\cloud\player\CloudPlayerManager;

final class PlayerTransferPacket extends CloudPacket implements ClientboundPacket, CloudboundPacket {

    public function __construct(
        private string $player = "",
        private string $server = ""
    ) {}

    public function handle(ServerClient $client): void {
        if (($player = CloudPlayerManager::getInstance()->get($this->player)) !== null) {
            $player->getCurrentProxy()?->sendPacket($this);
        }
    }

    public function encodePayload(PacketData $packetData): void {
        $packetData->writeAll($this->player, $this->server);
    }

    public function decodePayload(PacketData $packetData): void {
        $packetData->readAll($this->player, $this->server);
    }

    public function getPlayer(): string {
        return $this->player;
    }

    public function getServer(): string {
        return $this->server;
    }

    public static function create(string $player, string $server): self {
        return new self($player, $server);
    }
}

File: cloud-network/src/main/java/de/pocketcloud/network/packet/impl/PlayerTransferPacket.java
package de.pocketcloud.network.packet.impl;

import de.pocketcloud.api.network.packet.AuthenticatedPacket;
import de.pocketcloud.api.network.packet.ClientboundPacket;
import de.pocketcloud.api.network.packet.CloudboundPacket;
import de.pocketcloud.api.network.packet.data.IPacketData;
import de.pocketcloud.network.packet.CloudPacket;
import lombok.Getter;
import lombok.NoArgsConstructor;

@NoArgsConstructor
@Getter
public final class PlayerTransferPacket extends CloudPacket implements ClientboundPacket, CloudboundPacket, AuthenticatedPacket {

    private String player;
    private String server;

    public PlayerTransferPacket(String player, String server) {
        this.player = player != null ? player : "";
        this.server = server != null ? server : "";
    }

    @Override
    public void encodePayload(IPacketData packetData) {
        packetData.writeAll(player, server);
    }

    @Override
    public void decodePayload(IPacketData packetData) {
        this.player = packetData.readString();
        this.server = packetData.readString();
    }

    public static PlayerTransferPacket create(String player, String server) {
        return new PlayerTransferPacket(player, server);
    }
}
